pkg/subctl/cmd/join.go

```diff
@@ -419,6 +422,10 @@ func populateSubmarinerSpec(subctlData *datafile.SubctlData, netconfig globalnet
 		crClusterCIDR = netconfig.ClusterCIDR
 	}
 
+	if customDomains == nil {
+		customDomains = *subctlData.CustomDomains
```

Comment:
This is breaking e2e!

Reply:
Ah, I suspect you need to handle older data files where `CustomDomains` ends up `nil`.

Reply:
> Ah, I suspect you need to handle older data files where `CustomDomains` ends up `nil`.

Done, hope this fixes it.